crates/bevy_render/src/render_component.rs

```diff
@@ -111,14 +111,19 @@ fn prepare_uniform_components<C: Component>(
     C: AsStd140 + Clone,
 {
     component_uniforms.uniforms.clear();
-    for (entity, component) in components.iter() {
-        commands
-            .get_or_spawn(entity)
-            .insert(DynamicUniformIndex::<C> {
-                index: component_uniforms.uniforms.push(component.clone()),
-                marker: PhantomData,
-            });
-    }
+    let entities = components
+        .iter()
+        .map(|(entity, component)| {
+            (
+                entity,
+                (DynamicUniformIndex::<C> {
+                    index: component_uniforms.uniforms.push(component.clone()),
+                    marker: PhantomData,
+                },),
+            )
+        })
+        .collect::<Vec<_>>();
```

Comment:
Try using a `Local<Vec<...>>` here and clearing every frame instead of allocating a new `Vec`.

Alternatively, insert_or_spawn_batch takes an iterator. You could pass the iterator directly into it instead of collecting into a `Vec` here.

Reply:
@cart @mockersf - what do you think about this? The downside of `Local<Vec<T>>` is that the memory is held between system executions. The upside is that you don't have to re-allocate every frame.